### PR TITLE
Enable "Files" view to list actual files and dirs

### PR DESCRIPTION
**Pull request by hyunsik-yoon**
With this commit, "Files" view now shows the actual list of
files and dirs.

parent issue: https://github.com/Samsung/ONE-vscode/issues/330
draft: https://github.com/Samsung/ONE-vscode/pull/297
after: #371 

ONE-vscode-DCO-1.0-Signed-off-by: Hyun Sik Yoon <hyunsik.yoon.1024@gmail.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -143,8 +143,10 @@
   "devDependencies": {
     "@types/chai": "^4.3.0",
     "@types/glob": "^7.1.3",
+    "@types/mkdirp": "^1.0.2",
```
Comment (hyunsik-yoon) [inline package.json:146]: this package is needed by `import * as mkdirp from 'mkdirp';`

File: package.json [hunk 2]
```diff
@@ -143,8 +143,10 @@
...
     "@types/mocha": "^9.1.0",
     "@types/node": "^12.11.7",
+    "@types/rimraf": "^3.0.2",
```
Comment (hyunsik-yoon) [inline package.json:149]: this package is needed by `import * as rimraf from 'rimraf';`

**Comment by hyunsik-yoon** [comment]
I added a test dir at `res/modelDir` which contains one `tflite` file 
which can be used as a source of our tests.
In this PR, this file is used to show better communication for reviewer.

Note that although `tflite` is a binary file, the size is small. I think that we're OK with this file.

**Comment by hyunsik-yoon** [comment]
After entering directory which model files are stored,

image

Next screen will be 

image

File: src/Views/FileTreeView.ts [hunk 3]
```diff
@@ -134,13 +133,13 @@ export function exists(path: string): Promise<boolean> {
 
 export function rmrf(path: string): Promise<void> {
   return new Promise<void>((resolve, reject) => {
-    rimraf(path, error => handleResult(resolve, reject, error, void 0));
+    rimraf.default(path, error => handleResult(resolve, reject, error, void 0));
```
Comment (hyunsik-yoon) [inline src/Views/FileTreeView.ts:136]: Original line shows compilation error, probably due to `rimraf` version issue?

File: src/Views/FileTreeView.ts [hunk 4]
```diff
@@ -134,13 +133,13 @@ export function exists(path: string): Promise<boolean> {
...
   });
 }
 
 export function mkdir(path: string): Promise<void> {
   return new Promise<void>((resolve, reject) => {
-    mkdirp(path, error => handleResult(resolve, reject, error, void 0));
+    mkdirp.default(path);
```
Comment (hyunsik-yoon) [inline src/Views/FileTreeView.ts:142]: ditto

File: src/Views/FileTreeView.ts [hunk 5]
```diff
@@ -354,23 +361,27 @@ export class FileSystemProvider implements vscode.TreeDataProvider<Entry>,
         element.type === vscode.FileType.Directory ? vscode.TreeItemCollapsibleState.Collapsed :
                                                      vscode.TreeItemCollapsibleState.None);
     if (element.type === vscode.FileType.File) {
+      // TODO Change command to proper command, considering file types
       treeItem.command = {
-        command: 'fileExplorer.openFile',
+        command: 'vscode.open',
```
Comment (hyunsik-yoon) [inline src/Views/FileTreeView.ts:366]: This command opens an vscode default editor.

File: src/Views/FileTreeView.ts [hunk 6]
```diff
@@ -354,23 +361,27 @@ export class FileSystemProvider implements vscode.TreeDataProvider<Entry>,
...
         title: 'Open File',
         arguments: [element.uri],
       };
-      treeItem.contextValue = 'file';
+
+      const filePath = element.uri.fsPath;
+      if (filePath.endsWith('.tflite') || filePath.endsWith('.pth') || filePath.endsWith('.onnx')) {
```
Comment (llFreetimell) [inline src/Views/FileTreeView.ts:372]: Maybe `.pb` , `.circle`, `keras-model`, `saved_model` should be added?

Comment (hyunsik-yoon) [reply src/Views/FileTreeView.ts:372]: `pb` and `circle` seems OK.
keras model or saved model is saved into directory, AFAIK. So I will add TODO to check if the dir is for such models.
Also, we have to show backend-specific files... later.
E.g., if the extension of the result of backend-target code is, `bar`,
`bar` should be checked and, e.g., `treeItem.contextValue = targetCode` should be called. 😮

File: src/Views/FileTreeView.ts [hunk 7]
```diff
@@ -354,23 +361,39 @@ export class FileSystemProvider implements vscode.TreeDataProvider<Entry>,
...
-        command: 'fileExplorer.openFile',
+        command: 'vscode.open',
         title: 'Open File',
         arguments: [element.uri],
       };
-      treeItem.contextValue = 'file';
+
+      const filePath = element.uri.fsPath;
+
+      const isCompilableFile = (filePath: string) => {
```
Comment (YongseopKim) [inline src/Views/FileTreeView.ts:373]: In the future, this could be extracted as a util function. 👍

**Review by YongseopKim — APPROVED** [review]
LGTM

**Review by llFreetimell — APPROVED** [review]
LGTM

**Pull request merged**